pkg/scheduler/framework/runtime/registry.go

```diff
@@ -43,3 +47,38 @@ func (r Registry) Merge(in Registry) error {
 	}
 	return nil
 }
+
+// FactoryNames returns all known plugin names
+func (r Registry) FactoryNames() []string {
+	return sets.StringKeySet(r).List()
+}
+
+// Filter out the disabled plugin
+func (r Registry) Filter(names []string) Registry {
```

Comment:
Can we add a ut for this method?

I have a question: if the plugins is `-foo,*`, dose the result contain `foo`?

Reply:
> Can we add a ut for this method?

ok.

> dose the result contain foo?

no, the result is `*`

Reply:
If so, the behavior of `--plugins` will be different from controllers. I think that the writing habits should be consistent:

https://github.com/karmada-io/karmada/blob/master/docs/userguide/configure-controllers.md#configure-karmada-controllers

Reply:
ok

Reply:
We have a similar logic at `--controllers` in `karmada-controller-manager`, which is used another way:
https://github.com/karmada-io/karmada/blob/821a18542e57ae5ecf5fdc44bd3fa14c586f4a66/pkg/controllers/context/context.go#L76-L96.

I'm not sure if we can share some code here, will look at it later.